packages/owl-bot-post-processor/src/helpers.ts

```diff
@@ -0,0 +1,281 @@
+    }
+    return {
+      conclusion,
+      summary,
+      text,
+    };
+  } catch (err) {
+    logger.error(err.message);
+    return {
+      conclusion: 'failure',
+      summary: 'unknown build failure',
+      text: 'unknown build failure',
+    };
+  }
+}
+
+async function waitForBuild(
+  projectId: string,
+  id: string,
+  client: CloudBuildClient
+) {
+  for (let i = 0; i < 120; i++) {
+    const [build] = await client.getBuild({projectId, id});
+    if (build.status !== 'WORKING' && build.status !== 'QUEUED') {
+      return build;
+    }
+    // Wait a few seconds before checking the build status again:
+    await new Promise(resolve => {
+      setTimeout(() => {
+        return resolve(undefined);
+      }, 5000);
+    });
+  }
+  throw Error(`timed out waiting for buuild ${id}`);
+}
+
+export async function getHeadCommit(
+  owner: string,
+  repo: string,
+  pr: number,
+  octokit: OctokitType
+): Promise<Commit | undefined> {
+  // If a PR has more than 100 updates to it, we will currently have issues
+  // in practice this should be rare:
+  const {data: commits} = await octokit.pulls.listCommits({
+    owner,
+    repo,
+    pull_number: pr,
+    per_page: 100,
+  });
+  const headCommit = commits[commits.length - 1];
+  if (!headCommit) return;
+  else return headCommit as Commit;
+}
+
+export async function createCheck(
+  args: CheckArgs,
+  octokit?: OctokitType,
+  logger = console
+) {
+  if (!octokit) {
+    octokit = await getAuthenticatedOctokit({
+      'pem-path': args['pem-path'],
+      'app-id': args['app-id'],
+      installation: args.installation,
+    });
+  }
+  const [owner, repo] = args.repo.split('/');
+  const headCommit = await getHeadCommit(owner, repo, Number(args.pr), octokit);
+  if (!headCommit) {
+    logger.warn(`no commit found for PR ${args.pr}`);
+    return;
+  }
+  await octokit.checks.create({
+    owner,
+    repo,
+    name: 'OwlBot Post Processor',
+    summary: args.summary,
+    head_sha: headCommit.sha as string,
+    conclusion: args.conclusion,
+    output: {
+      title: args.title,
+      summary: args.summary,
+      text: args.text,
+    },
+  });
+}
+
+export async function getToken(
+  pemPath: string,
+  appId: number,
+  installation: string
+): Promise<Token> {
+  const privateKey = await readFileAsync(pemPath);
+  const payload = {
+    // issued at time
+    iat: parseInt('' + Date.now() / 1000),
+    // JWT expiration time (10 minute maximum)
+    exp: parseInt('' + Date.now() / 1000 + 10 * 60),
+    // GitHub App's identifier
+    iss: appId,
+  };
+  const jwt = sign(payload, privateKey, {algorithm: 'RS256'});
+  const resp = await request({
+    url: accessTokenURL(installation),
+    method: 'POST',
+    headers: {
+      Authorization: `Bearer ${jwt}`,
+      Accept: 'application/vnd.github.v3+json',
+    },
+  });
+  if (resp.status !== 201) {
+    throw Error(`unexpectedd response http = ${resp.status}`);
```

Comment:
"unexpectedd"

Reply:
I wanted to draw attention to just how **unexpectedd** this was.